### PR TITLE
Add Security Logging Plugin.

### PR DESCRIPTION
**Pull request by artivis**
# Add the Security Logging Plugin.
Currently the plugin logs only to files but ready for a first round of review.

Todo:
- [x] use the new [security logging macro](https://github.com/eProsima/Fast-RTPS/pull/1125/files#diff-674db8307f1595beddc8feca64bf8f8dR232) everywhere necessary
- [ ] more testing

Related PR:
- https://github.com/ros2/rmw_fastrtps/pull/361
- sros2 [coming soon]

Note:
- logging over-DDS will be implemented in a subsequent pull request

Summary:
* add security logging to file as per dds-spec
* add set/get logger to security plugin base classes
* logging integration to SecurityManager
* add SECURITY_LOGGING macros
* logging macro [show-case](https://github.com/eProsima/Fast-RTPS/pull/1125/files#diff-463177235a44520dd78a447dfb4fc12dR1105)

Signed-off-by: artivis <jeremie.deray@canonical.com>

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1757/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1514/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1749/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1774/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1527/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1768/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1779/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1531/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1773/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1783/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1535/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1777/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1785/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1537/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1779/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1786/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1538/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1780/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1787/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1539/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1781/)

**Comment by artivis** [comment]
Hi @richiware (@richiprosima),
I'm not sure why is this PR not compiling in CI whereas it does on my machine. The error is a bit unexpected. Do you have any clue of what's going on and/or how I could reproduce that?
Thanks in advance.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1810/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1558/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1804/)

**Comment by artivis** [comment]
> Greate job!! There are some coding-style issues. Maybe you might use this [uncrustify config file](https://github.com/richiware/cppstyle/blob/master/uncrustify.cfg).

Thanks @richiware. The code style is not homogeneous throughout the project. If I apply the config you provide only to my changes, some files may end up with two different styles. On the other hand applying it globally to the files I edited will result in unnecessary long diff. Do you want me to apply the conf only to my changes and have some files with two styles or only where it seems to make sense ?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1820/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1567/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1814/)

**Comment by artivis** [comment]
> Build status:
> 
> * Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1820/)
> * Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1567/)
> * Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1814/)

The test failures seem to be only related to timeouts. If anyone can confirm?
Despite this, I would rate this PR as ready for first reviews.

File: include/fastdds/rtps/security/logging/LogOptions.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+/*!
+ * @file LogOptions.h
+ */
+#ifndef _FASTDDS_RTPS_SECURITY_LOGGING_LOGOPTIONS_H_
+#define _FASTDDS_RTPS_SECURITY_LOGGING_LOGOPTIONS_H_
+
+#include <string>
+
+#include "fastdds/rtps/security/logging/LoggingLevel.h"
```
[the remark this answers is not in the document]
Comment (artivis) [reply include/fastdds/rtps/security/logging/LogOptions.h]: Done in 5e099c1dd43b64aba727edf7689ea185697b1d5a

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1822/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1569/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1816/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1835/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1582/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1829/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1842/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1588/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1836/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1844/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1590/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1838/)

**Comment by richiware** [comment]
This PR is ready to be merged. It only needs to be fixed the failure with DCO plugin.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1883/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1623/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1876/)

**Comment by MiguelCompany** [comment]
@richiware I manually accepted the DCO, as the offending commits were clearly from @artivis . This is ready to be merged

**Comment by richiware** [comment]
@artivis Could you confirm it is ready for merging? or is something still pending?

**Comment by artivis** [comment]
@richiware this PR is ready indeed :+1:.

**Pull request merged**